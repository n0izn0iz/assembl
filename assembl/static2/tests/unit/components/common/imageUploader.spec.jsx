import React from 'react';
import renderer from 'react-test-renderer';

import ImageUploader from '../../../../js/app/components/common/imageUploader';

describe('ImageUploader component', () => {
  it('should render an image uploader', () => {
    const handleChangeSpy = jest.fn(() => {});
    const component = renderer.create(<ImageUploader imgUrl="http://www.example.com/foobar.png" handleChange={handleChangeSpy} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an image uploader without preview', () => {
    const handleChangeSpy = jest.fn(() => {});
    const file = new File([''], 'foobar.png');
    const component = renderer.create(<ImageUploader imgUrl={file} handleChange={handleChangeSpy} withPreview={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});