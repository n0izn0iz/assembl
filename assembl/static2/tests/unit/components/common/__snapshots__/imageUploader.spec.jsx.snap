// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploader component should render an image uploader 1`] = `
<div>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className={undefined}
      style={undefined}
    >
      Upload Button
    </span>
  </button>
  <div
    className="preview"
  >
    <img
      alt="preview"
      src="http://www.example.com/foobar.png"
    />
  </div>
  <div
    className="preview-title"
  >
    
  </div>
  <input
    className="hidden"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`ImageUploader component should render an image uploader without preview 1`] = `
<div>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className={undefined}
      style={undefined}
    >
      Upload Button
    </span>
  </button>
  <div
    className="preview-title"
  >
    
  </div>
  <input
    className="hidden"
    onChange={[Function]}
    type="file"
  />
</div>
`;
